source/MaterialXRuntime/Commands/MetadataCommands.cpp

```diff
@@ -0,0 +1,50 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+#include <MaterialXRuntime/Commands/MetadataCommands.h>
+
+#include <MaterialXRuntime/Private/Commands/PvtSetMetadataCmd.h>
+#include <MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.h>
+
+#include <MaterialXRuntime/Private/PvtApi.h>
+
+#include <MaterialXRuntime/RtTypeDef.h>
+
+namespace MaterialX
+{
+
+namespace RtCommand
+{
+
+void setMetadataFromString(RtPrim& src, const RtToken& metadataName, const string& metadataValue, RtCommandResult& result)
```

Comment:
I think this command should be named "setMetadata", to be consistent with setAttribute. I know there is also a setAttributeFromString but that command supports any datatype value being set from string (looking at the datatype of the attribute passed in). So if this is the intent here we must also pass in an RtToken for the datatype to be set.

But if your requirements right now is only to set string typed metadata then a command "setMetadata" for string data is enough for now, and we can extend with other setMetadata for other datatypes later, just as we have setAttribute for many datatypes.